app/assets/javascripts/school_administrator_dashboard/dashboard_components/discipline_dashboard/SchoolDisciplineDashboard.js

```diff
@@ -0,0 +1,184 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import DashboardHelpers from '../DashboardHelpers';
+import StudentsTable from '../StudentsTable';
+import DashboardBarChart from '../DashboardBarChart';
+import DateSlider from '../DateSlider';
+
+class SchoolDisciplineDashboard extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      startDate: DashboardHelpers.schoolYearStart(),
+      selectedChart: {
+        type: "location",
+        data: this.props.disciplineIncidentsByLocation,
+        title: "Incidents by Location"},
+      selectedCategory: null
+    };
+    this.setDate = (range) => {
+      this.setState({
+        startDate: moment.unix(range[0]).format("YYYY-MM-DD")
+      });
+    };
+    this.setStudentList = (highchartsEvent) => {
+      this.setState({selectedCategory: highchartsEvent.point.category});
+    };
+    this.resetStudentList = () => {
+      this.setState({selectedCategory: null});
+    };
+    this.selectChart = (event) => {
+      this.setState({selectedChart: this.updateChartSelection(event.target.value), selectedCategory: null});
+    };
+  }
+
+  filterIncidentDates(incidents) {
+    return incidents.filter((incident) => {
+      return moment(incident.occurred_at).isSameOrAfter(moment(this.state.startDate));
+    });
+  }
+
+  studentDisciplineIncidentCounts(incidentCategory) {
+    let studentDisciplineIncidentCounts = {};
+    const incidents = incidentCategory ? this.state.selectedChart.data[incidentCategory] : this.props.totalDisciplineIncidents;
+    this.filterIncidentDates(incidents).forEach((incident) => {
+      studentDisciplineIncidentCounts[incident.student_id] = studentDisciplineIncidentCounts[incident.student_id] || 0;
+      studentDisciplineIncidentCounts[incident.student_id]++;
+    });
+    return studentDisciplineIncidentCounts;
+  }
+
+  updateChartSelection(selectedChart) {
```

Comment:
These values are all a function of the value of `selectedChart`, so you don't need to store them as state.  They can be computed in `render` or one of the other methods it calls.  The overall idea is "state should have the minimal bits of actual state, and then compute everything else based on that."

Reply:
Good point @kevinrobinson.